File: 2nd Semester/Object Oriented Programming Lab/Lab 06/Question_03.java
//Hafsa Salman
//22K-5161
//Question no. 03

public class Question_03
{
    public static void main(String[] args)
    {
        System.out.println("Name: Hafsa Salman");
        System.out.println("Roll no. 22K-5161");
        System.out.println("Question no. 03");
        System.out.println();

        Car car = new Car();

        car.Display();
    }
}

class Vehicle
{
    private int speed;
    private String color;
    private int wheels;

    public int getSpeed()
    {
        return speed;
    }

    public void setSpeed(int speed)
    {
        this.speed = speed;
    }

    public String getColor()
    {
        return color;
    }

    public void setColor(String color)
    {
        this.color = color;
    }

    public int getWheels()
    {
        return wheels;
    }

    public void setWheels(int wheels)
    {
        this.wheels = wheels;
    }

    public Vehicle()
    {
        this.speed = 0;
        this.color = "White";
        this.wheels = 4;
    }
    public Vehicle (int speed, String color,int wheels)
    {
        this.speed = speed;
        this.color = color;
        this.wheels = wheels;
    }
}

class MotorVehicle extends Vehicle
{
    private String licenseplate;

    public void setLicenseplate(String licenseplate)
    {
        this.licenseplate = licenseplate;
    }

    public String getLicenseplate()
    {
        return licenseplate;
    }

    public MotorVehicle()
    {
        super();
    }

    public MotorVehicle(String licenseplate)
    {
        this.licenseplate = "ABC-123";
    }

    public MotorVehicle(int speed, String color, int wheels, String licenseplate)
    {
        super(speed, color, wheels);
        this.licenseplate = licenseplate;
    }
}

class Car extends MotorVehicle
{
    private int door;

    public int getDoor()
    {
        return door;
    }

    public void setDoor(int door)
    {
        this.door = door;
    }

    public Car()
    {
        super();
    }

    public void Display()
    {
        System.out.println("Speed: " + getSpeed());
        System.out.println("Colour: " + getColor());
        System.out.println("No. of wheels: " + getWheels());
        System.out.println("License Plate: " + getLicenseplate());
        System.out.println("No. of doors: " + door);
    }
}

File: 3rd Semester/Data Structures and Algorithms Labs/DS Lab (Miss Bushra)/Lab 01_ Basic C++/Question_03.cpp
//Hafsa Salman
//22K-5161
//Question no. 03

#include<iostream>
using namespace std;

int main ()
{
	cout << "Name: Hafsa Salman" << endl;
	cout << "Roll no. 22K-5161" << endl;
	cout << "Question no. 03" << endl;
	
	int arr[5] = {10, 20, 30, 40, 50};
	int *ptr = arr;
	
	cout << "Array: \n";
	for (int i=0; i<5; i++)
	{
		cout << arr[i] << "\t";
	}
	
	ptr = arr;
	
	for (int i=0; i<5; i++)
	{
		*ptr *=2;
		
		ptr++;
	}
	
	cout << "\n\nArray after increment: \n";
	for (int i=0; i<5; i++)
	{
		cout << arr[i] << "\t";
	}
}